### PR TITLE
test: Fix `TestNodeStageVolume` on Windows

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**

**What this PR does / why we need it**:
Fix `TestNodeStageVolume` on Windows

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by mayankshah1607** [comment]
@andyzhangx I have tried to take a different approach for making the tests compatible across Windows as well as Linux. I have introduced a `TestError` struct, that holds errors for Windows and Linux, and assert on them accordingly by checking the OS

File: pkg/smb/nodeserver_test.go [hunk 1]
```diff
@@ -61,52 +57,72 @@ func TestNodeStageVolume(t *testing.T) {
...
 				VolumeContext:    volContext,
 				Secrets:          secrets},
-			expectedErr: status.Errorf(codes.Internal, "volume(vol_1##) mount \"test_source\" on \"./error_mount_sens_source\" failed with fake MountSensitive: target error"),
+			expectedErr: testutil.TestError{
+				DefaultError: status.Errorf(codes.Internal,
+					fmt.Sprintf("volume(vol_1##) mount \"test_source\" on \"%s\" failed with fake MountSensitive: target error",
+						errorMountSensSource)),
+				// Not a desired error. This will need a better fix
+				WindowsError: fmt.Errorf("prepare stage path failed for %s with error: could not cast to csi proxy class", errorMountSensSource),
+			},
```
Comment (mayankshah1607) [inline pkg/smb/nodeserver_test.go:114]: This is not the desired error. While we would expect a target error, this test fails in Windows due to some type casting. We'll need a fix for the long term. I' not sure why this is happening. Any ideas?

File: pkg/smb/nodeserver_test.go [hunk 2]
```diff
@@ -61,52 +57,72 @@ func TestNodeStageVolume(t *testing.T) {
...
 			desc: "[Success] Valid request",
 			req: csi.NodeStageVolumeRequest{VolumeId: "vol_1##", StagingTargetPath: sourceTest,
 				VolumeCapability: &stdVolCap,
 				VolumeContext:    volContext,
 				Secrets:          secrets},
-			expectedErr: nil,
+			expectedErr: testutil.TestError{
+				// Not a desired error. This will need a better fix
+				WindowsError: fmt.Errorf("prepare stage path failed for %s with error: could not cast to csi proxy class", sourceTest),
+			},
```
Comment (mayankshah1607) [inline pkg/smb/nodeserver_test.go:125]: Similar type casting error here on Windows

File: test/utils/testutil/testutil.go [hunk 3]
```diff
@@ -16,9 +16,59 @@ limitations under the License.
...
+}
+
+// GetWorkDirPath returns the path to the current working directory
+func GetWorkDirPath(dir string, t *testing.T) string {
+	path, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("failed to get working directory: %s", err)
+	}
+	var sep string
+	if isWindows() {
```
Comment (andyzhangx) [inline test/utils/testutil/testutil.go:68]: use this `os.PathSeparator`

File: pkg/smb/fake_mounter_test.go [hunk 4]
```diff
@@ -25,6 +25,9 @@ import (
 )
 
 func TestMount(t *testing.T) {
+	targetTest := "./target_test"
+	sourceTest := "./source_test"
```
Comment (mayankshah1607) [inline pkg/smb/fake_mounter_test.go:29]: Whenever these tests are refactored, we'll similarly change the hard coded paths to `testutil.GetWorkDirPath`

File: pkg/smb/nodeserver_test.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/smb/nodeserver_test.go:114]: add a `// todo: `

File: pkg/smb/nodeserver_test.go (hunk 2 above)
Comment (andyzhangx) [reply pkg/smb/nodeserver_test.go:125]: add a `// todo: `

File: pkg/smb/fake_mounter_test.go (hunk 4 above)
Comment (andyzhangx) [reply pkg/smb/fake_mounter_test.go:29]: anything you want to do in this PR?

Comment (mayankshah1607) [reply pkg/smb/fake_mounter_test.go:29]: Nope, I think we can merge this one. I will open a follow up PR for refactoring remaining tests on this repo, as well as on the azurefile and azure disk repos

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**